Fixing spacing around chaise buttons

PoornimaVKrishnan opened the pull request:
This PR will address [the issue](https://github.com/informatics-isi-edu/chaise/issues/2349)

Summary of changes:

The aim of this PR change is to fix the inconsistency in button spacing issue.

With this PR we have center aligned the buttons using display: 'inline-flex'.

Also for view details action button a height is given and padding is removed to fix the alignment
The deployed version is [here](https://dev.gudmap.org/~povijaya/chaise/record/#2/Common:Collection/RID=14-4KG6)

RFSH commented on src/assets/scss/helpers.scss:
```diff
@@ -89,6 +88,10 @@
   -ms-user-select: none;
   white-space: nowrap;
+  // ficxing button spacing issue. Fix- center aligning the buttons
+  display: inline-flex !important;
```
You mentioned that you're using `!important` here to override the displays that we've defined in some buttons. 

I was thinking about it, and Instead of `!important` let's just remove the display definition from those other places. We should keep the rules in here as general as possible and should allow other places to be able to override them.

RFSH commented on src/assets/scss/helpers.scss:
```diff
@@ -89,6 +88,10 @@
   -moz-user-select: none;
   -ms-user-select: none;
   white-space: nowrap;
+  // ficxing button spacing issue. Fix- center aligning the buttons
```
Fix the typo

jrchudy commented on src/assets/scss/helpers.scss:
```diff
@@ -89,6 +88,10 @@
+  // ficxing button spacing issue. Fix- center aligning the buttons
+  display: inline-flex !important;
+  align-items: center;
+  justify-content: center;
```
Add `line-height: normal`. Add a comment explaining what `normal` does and why it works here. Or try other values and explain why that might be better.

Adding `line-height` here will allow for the `<span>` tags to inherit the value and use it for the text part of the button.

jrchudy reviewed (changes requested):
I tested these changes myself in chrome on window and still see the issue. I added a comment about adding `line-height` to the `chaise-btn` styles so the children of the button (usually `<span>` tags) will use this property instead of what bootstrap is defining.

I'm unsure of which value line-height should be but using `normal` seemed to be the best so it is consistent for our different button sizes too.

RFSH reviewed (changes requested):
Related to positing the "action buttons" in recordset table, ideally we want the buttons to be aligned with the first line of the other cells. We also want to preserve space as much as we can, so we still want to reduce the bottom padding and you should not remove the rule in line 164. 

- In Chrome the alignment looks better but I still think it's a few pixel off from the middle. 
![Chrome](https://github.com/informatics-isi-edu/chaise/assets/2129750/308282ca-a784-46a7-bc01-922d35262845)

- In Firefox the added height is actually making the view button to be way above the rest. 
![Firefox](https://github.com/informatics-isi-edu/chaise/assets/2129750/45221c6b-89fe-4e95-a507-aed6fd9eb33c)

For chrome, I think if you use `padding: 3px 0 0 0;` in line 164 instead of 5, the alignment would be much better.

Make sure you're testing your changes on Chrome, Firefox, and Safari. This might be related to line-height that Josh mentioned. Maybe different browsers are computing different line-heights?

PoornimaVKrishnan commented:
@RFSH I have addressed all comments. Please have a look. 
[here](https://dev.gudmap.org/~povijaya/chaise-2/record/#2/Common:Collection/RID=14-4KG6)
I tested on chrome and firefox in different zooming levels

RFSH commented:
@PoornimaVKrishnan Please be more attentive. I'm assuming you were only focused on the create/copy/edit/delete button. But the "Export" and "Permalink" buttons in recordset now show up in two lines.

Also did you look at my comment? I asked you to try to fix the alignment of the action buttons but you only reverted the changes that you did. Like I said try `3px` padding and see if that is better than what we have or not. In our current version the action buttons are a few pixel lower than they should be.

PoornimaVKrishnan commented:
> @PoornimaVKrishnan Please be more attentive. I'm assuming you were only focused on the create/copy/edit/delete button. But the "Export" and "Permalink" buttons in recordset now show up in two lines.
> 
> Also did you look at my comment? I asked you to try to fix the alignment of the action buttons but you only reverted the changes that you did. Like I said try `3px` padding and see if that is better than what we have or not. In our current version the action buttons are a few pixel lower than they should be.

I fixed the issue. Added comments. `3px` looked better

PoornimaVKrishnan commented on src/assets/scss/_record.scss:
```diff
@@ -43,6 +43,11 @@
+              & > * {
+                display: inline-flex;
+              }
             }
```
added this to override title-button class, giving the change there was causing the export and permlink button to come in two lines

PoornimaVKrishnan commented on src/assets/scss/_recordset-table.scss:
```diff
@@ -157,21 +157,21 @@
-    // since we're increasing the size of icons, we should decrease the padding
-    padding: 5px 0 0 0;
-
+    padding: 3px 0 0 0;
```
Decided to go with `3px` as the alignment looked fine with other columns

PoornimaVKrishnan commented on src/assets/scss/_recordset-table.scss:
```diff
@@ -157,21 +157,21 @@
     .chaise-view-details {
-      font-size: 1.1em;
+      font-size: 1em;
     }
```
edit pencil icon was few pixels off from view-details. reducing the font-size looks better in all zoom levels and browsers

PoornimaVKrishnan commented on src/assets/scss/_recordset-table.scss:
```diff
@@ -157,21 +157,22 @@
-    padding: 5px 0 0 0;
-
+    padding: 3px 0 0 0;
+    line-height: 1.5;
```
added line-height as it was not aligning in windows when i tested in saucelabs. this is the closest alignment i could get for the action buttons

RFSH commented:
@PoornimaVKrishnan  The main change in this PR is addition of `display: inline-flex` to buttons. We should not override this rule to try to solve other issues. Because of that change, the "Export" and "Permalink" button labels in recordset were misaligned. So I pushed a fix for this. To fix this we just needed to add `display: inline-block` to the `.export-menu.dropdown`.

Also while the action buttons spacing were a lot better than what we had before, the alignment was still ugly in the one-line case. This was more obvious on the facet and foreign key popups. So I also pushed some changes related to that.

@jrchudy To me this PR is ready to be merged. Can you please review it again?

RFSH commented:
After discussing this with @hongsudt, we decided to reduce the size of action buttons. This was done because in the single select mode the button was taking up the whole available height and everything was very close to each other.

While doing this change I also fixed the display of checkboxes on Firefox. Firefox doesn't like the way we were attaching `:before` and `:after` to input, so I had to move it to `label` instead. Now all browsers will display the same checkbox.
